runtime/ms-rest/lib/webResource.js

```diff
@@ -163,7 +163,7 @@ class WebResource {
       }
       let baseUrl = options.baseUrl;
       let url = baseUrl + (baseUrl.endsWith('/') ? '' : '/') + (options.pathTemplate.startsWith('/') ? options.pathTemplate.slice(1) : options.pathTemplate);
-      let segments = url.match(/({\w*\s*\w*})/ig);
+      let segments = url.match(/({\W*\S*\W*})/ig);
```

Comment:
I'm not sure I understand the fix. With this change, a text like `{foo bar}` that used to be a match for the old pattern, will no longer be a match. Is that intentional, why?

In all, this change can be summarized as so :
1. change looking for any word character to any non word character
2. change looking for a space to any non-whitespace character

I don't understand this change, feel free to drop by and talk to me 😃